RoddyCore/src/de/dkfz/roddy/execution/io/LocalExecutionService.groovy

```diff
@@ -38,25 +51,72 @@ class LocalExecutionService extends ExecutionService {
         return true
     }
 
-    // This method actually overrides a base class. But if we keep the @Override, the Groovy (or Java) compiler constantly
-    // claims, that the method does not override it's base method.
-    // That is, why we keep it in but only as a comment.
-//     @Override
-    protected ExecutionResult _execute(String command, boolean waitFor, boolean ignoreErrors, OutputStream outputStream = null) {
+    /** Read from an InputStream asynchronously using the executorService.
+        May throw an UncheckedIOException.
+     **/
+    private CompletableFuture<List<String>> asyncReadStringStream(
+            InputStream inputStream,
+            ExecutorService executorService = executorService) {
+        return CompletableFuture.supplyAsync({
+            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))
+            reader.lines().toArray() as List<String>
+        } as Supplier<List<String>>, executorService)
+    }
+
+    /**
+     * @param command        The command to execute. Note, this is always executed with Bash as the command interpreter.
+     * @param waitFor        Whether to do synchronous (true) or asynchronous execution. Asynchronous execution
+     *                       will be done on the class-internal thread-pool. Don't do this with mutually blocking
+     *                       processes.
+     * @param ignoreErrors   Ignored. No execution errors are reported. This is really nasty, but that's the way it
+     *                       is in the moment that doesn't break dependencies. Instead the method logs errors to the
+     *                       logging output.
+     * @param outputStream   Ignored.
+     * @return
+     */
+    @Override
+    protected ExecutionResult _execute(String command,
+                                       boolean waitFor,
+                                       boolean ignoreErrors,
+                                       OutputStream outputStream = null) {
         if (waitFor) {
             return LocalExecutionHelper.executeCommandWithExtendedResult(command, outputStream)
         } else {
-            Thread.start {
-                command.execute()
-            }
-            return new ExecutionResult(true, 0, [], "")
+            ProcessBuilder processBuilder = new ProcessBuilder(["bash", "-c", command])
```

Comment:
The old implementation used `String.execute(String)`, which eventually refers to `Runtime.exec(String command, String[] end, File dir)`. This latter method tokenizes the command using the delimiters space, tab, newline, carriage-return, form-feed. Bash, by contrast, uses the delimiters space, tab, newline by default (the `$IFS` variable), which, however, in this context here boils down to the same, because here only spaces are relevant token-delimiters. This means that in our context, `command.execute()` and `bash -c ...` called via `ProcessBuilder` will do the same.